addons/website_blog/static/src/js/blog_tag_add.js

```diff
@@ -0,0 +1,201 @@
+            return true;
+        });
+        this.rerender = true;
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    addTag: function (previewMode, widgetValue, params) {
+        const addedTagId = parseInt(widgetValue);
+        this.availableTags = this.availableTags.filter(tag => {
+            if (tag.id === addedTagId) {
+                this.tagIds.push({
+                    'id': tag.id,
+                    'name': tag.name,
+                });
+                this.addList.push(tag);
+                this._addTagPreview(tag);
+                return false;
+            }
+            return true;
+        });
+        this.rerender = true;
+    },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    updateUI: async function () {
+        // If we want to rerender the xml we need to avoid the updateUI
+        // as they are asynchronous and the ui might try to update while
+        // we are building the UserValueWidgets.
+        if (this.rerender) {
+            this.rerender = false;
+            await this._rerenderXML();
+            return;
+        }
+        await this._super.apply(this, arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _renderCustomXML: async function (uiFragment) {
+        this.$tagList = $(uiFragment.querySelector('[data-name="tag_list"]'));
+        this.$select = $(uiFragment.querySelector('[data-name="add_tag_opt"] we-select'));
+        this.tagIds.forEach(tag => {
+            this.$tagList.append(qweb.render('website_blog.tagListItem', {
+                tag: tag,
+                isEditing: this.isEditing,
+            }));
+        });
+        this.availableTags.forEach(tag => {
+            this.$select.prepend(qweb.render('website_blog.tagSelectItem', {
+                tag: tag,
+                isEditing: this.isEditing,
+            }));
+        });
+    },
+    _addTagPreview: function (tag) {
+        const $tagList = $('.tag_list');
+        const $newTag = $(qweb.render('website_blog.tag_link', {tag}));
+        $tagList.append($newTag);
+    },
+    _removeTagPreview: function (tag) {
+       const $removedTag = $(`.${tag.name}-${tag.id}`);
+       $removedTag.remove();
+    },
+    _saveChanges: function () {
+        const proms = [];
+        this.addList.forEach(async tag => {
+            if (tag.id === 0) {
+                tag.id = await this._rpc({
+                    model: 'blog.tag',
+                    method: 'create',
+                    args: [{
+                        'name': tag.name,
+                    }],
+                });
+            }
+            proms.push(this._rpc({
```

Comment:
the post_ids could be done in the create rpc for new tag.
And this Rpc with a `else` condition